crates/ra_hir/src/code_model_impl/module.rs

```diff
@@ -125,15 +130,37 @@ impl Module {
         );
 
         let segments = &path.segments;
-        for name in segments.iter() {
+        for (idx, name) in segments.iter().enumerate() {
             let curr = if let Some(r) = curr_per_ns.as_ref().take_types() {
                 r
             } else {
                 return Ok(PerNs::none());
             };
             let module = match curr.resolve(db)? {
                 Def::Module(it) => it,
-                Def::EnumVariant(ev) => return Ok(PerNs::values(ev.def_id)),
+                Def::Enum(e) => {
+                    if let Some(suffix) = segments.get(idx + 1) {
+                        assert!(segments.len() == idx);
```

Comment:
I don't think we should panic here - we still need to work out error reporting, but at the moment we can just return `Ok(PerNS::none())`. Same for below.

Reply:
:+1: will do